src/puppetlabs/puppetdb/http/query.clj

```diff
@@ -153,3 +211,88 @@
    :post [(are-queries-different? req %)]}
   (restrict-query ["=" "title" title]
                   req))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Conversion/validation of query parameters
+
+(defn parse-boolean
+  "Parses `b` to a boolean unless it's already a boolean"
+  [b]
+  (if (instance? Boolean b)
+    b
+    (Boolean/parseBoolean b)))
+
+(defn count?
+  "If the original query contains the include_total w/o a a value, it
+  will show up here as nil.  We assume that in that case, the caller
+  intended to use it as a flag"
+  [query-map]
+  (let [total (get query-map :include_total ::not-found)]
+    (if (= total ::not-found)
+      false
+      (parse-boolean total))))
+
+(pls/defn-validated convert-query-params :- puppetdb-query-schema
+  "This will update a query map to contain the parsed and validated query parameters"
+  [{:keys [order_by limit offset] :as full-query}]
+  (-> full-query
+      (update-in [:order_by] parse-order-by')
+      (update-in [:limit] parse-limit')
+      (update-in [:offset] parse-offset')
+      (assoc :count? (count? full-query))
+      (dissoc :include_total)))
+
+(defn get-req->query
+  "Converts parameters of a GET request to a pdb query map"
+  [{:keys [params] :as req}]
+  (conj {:query (json/parse-strict-string (get params "query") true)
+         :order_by (json/parse-strict-string (get params "order_by") true)
+         :limit (get params "limit")
+         :offset (get params "offset")}
+        (let [include-total? (get params "include_total" ::not-found)]
+          (when (not= include-total? ::not-found)
+            [:include_total include-total?]))))
+
+(defn post-req->query
+  "Takes a POST body and parses the JSON to create a pdb query map"
+  [req]
+  (-> req
+      :body
+      clojure.java.io/reader
```

Comment:
Good question. Looking through the code, I don't think it does. I'll change this